kafka/client.go

```diff
@@ -16,11 +16,15 @@ type TopicMissingError struct {
 
 func (e TopicMissingError) Error() string { return e.msg }
 
+type void struct{}
+var member void
+
 type Client struct {
 	client        sarama.Client
 	kafkaConfig   *sarama.Config
 	config        *Config
 	supportedAPIs map[int]int
+	topics        map[string]void
```

Comment:
the underlying sarama client is VERY inefficient when testing for existence of topics. We need to maintain a map cache at this level to speed things up.